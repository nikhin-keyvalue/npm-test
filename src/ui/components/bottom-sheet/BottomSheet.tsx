import Preact, { h } from 'preact';
import { useState, useEffect } from 'preact/hooks';

import { BottomSheetTypes } from 'constants/common';
import { isMobileDevice } from 'ui/utils/common';

import './style.scss';

type BottomSheetProps = {
  isOpen: boolean;
  children: any;
  setBottomSheetOpen: (state: BottomSheetTypes | null) => void;
};

const BottomSheet: Preact.FunctionComponent<BottomSheetProps> = ({ isOpen = false, children, setBottomSheetOpen }) => {
  const [isModalOpen, toggleModalOpen] = useState(false);
  const [isSheetOpen, toggleSheetOpen] = useState(false);

  useEffect(() => {
    if (isOpen) {
      toggleModalOpen(isOpen);
      setTimeout(() => {
        toggleSheetOpen(isOpen);
      }, 50);
    } else {
      setTimeout(() => {
        toggleSheetOpen(isOpen);
      }, 50);

      setTimeout(() => {
        toggleModalOpen(isOpen);
      }, 250);
    }
  }, [isOpen]);

  return (
    <div
      className={`zamp-bottom-sheet-wrapper ${isModalOpen ? 'zamp-bottom-sheet-open' : ''}`}
      role='presentation'
      onClick={() => setBottomSheetOpen(null)}
    >
      <div
        className={`zamp-bottom-sheet-content ${isSheetOpen ? 'zamp-sheet-content-open' : 'zamp-sheet-content-close'} ${
          isMobileDevice() ? 'zamp-bottomsheet-mobile-padding' : ''
        }`}
        role='presentation'
        onClick={(e) => e.stopPropagation()}
      >
        {children}
      </div>
    </div>
  );
};

export default BottomSheet;
