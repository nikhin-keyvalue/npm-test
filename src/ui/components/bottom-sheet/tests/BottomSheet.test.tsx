import { h } from 'preact';
import { configure, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Adapter } from 'enzyme-adapter-preact-pure';

import BottomSheet from '../BottomSheet';

configure({ adapter: new Adapter() });

describe('Counter', () => {
  it('render components', () => {
    const wrapper = mount(
      <BottomSheet isOpen={true} setBottomSheetOpen={() => null}>
        {}
      </BottomSheet>
    );
    const tree = shallowToJson(wrapper);

    expect(tree).toMatchSnapshot();
  });
});
