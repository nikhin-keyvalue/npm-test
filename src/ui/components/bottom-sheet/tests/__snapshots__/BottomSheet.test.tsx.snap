// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter render components 1`] = `
<BottomSheet
  isOpen={true}
  setBottomSheetOpen={[Function]}
>
  <div
    className="zamp-bottom-sheet-wrapper zamp-bottom-sheet-open"
    onClick={[Function]}
    role="presentation"
  >
    <div
      className="zamp-bottom-sheet-content zamp-sheet-content-close "
      onClick={[Function]}
      role="presentation"
    />
  </div>
</BottomSheet>
`;
